s3fs/core.py

```diff
@@ -1814,14 +1814,49 @@ def __init__(
         self.append_block = False
 
         if "a" in mode and s3.exists(path):
-            loc = s3.info(path)["size"]
+            head = {
+                k: v
+                for k, v in self._call_s3(
+                    "head_object",
+                    self.kwargs,
+                    Bucket=bucket,
+                    Key=key,
+                    **version_id_kw(version_id),
+                    **self.req_kw,
+                ).items()
+                if k
+                in {
+                    "CacheControl",
```

Comment:
Do you please have a source for this list of keys? I would be pretty surprised at some of them if they were meant to be passed back as part of the upload.

Reply:
Added comments

Reply:
thank you